File: server/internal/models/plc.go
package models

import "time"

type PlcSensors struct {
	Timestamp         time.Time `json:"timestamp"`
	FlowRate          float64   `json:"flowRate"`          // m3/s
	PH                float64   `json:"pH"`                // 0–14
	Turbidity         float64   `json:"turbidity"`         // NTU
	DissolvedOxygen   float64   `json:"dissolvedOxygen"`   // mg/L
	Temperature       float64   `json:"temperature"`       // °C
}

File: client/src/types/plc.ts
export type PlcSensorReading = {
	timestamp: string;
	flowRate: number;
	pH: number;
	turbidity: number;
	dissolvedOxygen: number;
	temperature: number;
};

export type PlcStreamStatus = "connecting" | "open" | "closed" | "error";

export type StatusTone = "live" | "muted" | "warning";